flag_impl.go

```diff
@@ -13,47 +13,35 @@ type Value interface {
 	flag.Getter
 }
 
-// simple wrapper to intercept Value operations
-// to check for duplicates
-type valueWrapper struct {
-	value    Value
-	count    int
-	onlyOnce bool
+type boolFlag interface {
+	IsBoolFlag() bool
 }
 
-func (v *valueWrapper) String() string {
-	if v.value == nil {
-		return ""
-	}
-	return v.value.String()
+type fnValue struct {
+	fn     func(string) error
+	isBool bool
+	v      Value
 }
 
-func (v *valueWrapper) Set(s string) error {
-	if v.count == 1 && v.onlyOnce {
-		return fmt.Errorf("cant duplicate this flag")
+func (f fnValue) Get() any           { return f.v.Get() }
+func (f fnValue) Set(s string) error { return f.fn(s) }
+func (f fnValue) String() string {
```

Comment:
Are these function receivers intentionally not `*fnValue`?

Reply:
Not really. I tend to avoid pointer receivers if there is nothing to change in the struct. But yeah I'll change it